packages/grid/data-grid/src/tests/keyboard.DataGrid.test.tsx

```diff
@@ -92,9 +92,9 @@ describe('<DataGrid /> - Keyboard', () => {
     );
     const input = screen.getByTestId('custom-input');
     input.focus();
-    expect(getActiveCell()).to.equal('0-1');
+    expect(getActiveCell()).to.equal('0-2');
```

Comment:
This looks very wrong. The x and y axis have two different bases (zero base vs. one base) now.

@dtassone previously defended the idea to keep zero-base for the internal state. We have to choice between keeping a zero-base for the test assertions or to move to a 1-base to follow the aria convention. What should we do?

Rely on the internal convention, or on the aria convention? I would be leaning for the latter, as more intuitive.

Reply:
I would +1 for the aria so we can be agnostic to the implementation, it also seems more intuitive.

Reply:
Will wait for @dtassone's review. If considering aria based then we need to change `getActiveCell` implementation right?

Reply:
The index is 0 based because it is the index position in the array of columns. As always, arrays starts at 0.
So internally I think we should keep a 0 based index.

Then when we render cells, we can just add 1 for the aria-colindex and aria-rowindex.

They are one-based as they're used by screen readers.
https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Gridcell_role

For the tests, I think we should use the 0 based convention as it would be easier to map with the real internal position of a column or a row in the array of columns.

Therefore, `GetActiveCell` should use data-rowindex and data-colindex, so the first cell will be 0-0 instead of 1-1 or worse 0-1. IMO, it makes more sense for developers to be 0 based.
What do you think?

Reply:
There is no `data-colindex`. Should we add one in `GridCell`?

Reply:
> For the tests, I think we should use the 0 based convention as it would be easier to map with the real internal position of a column or a row in the array of columns.
Agree, I also vote for zero-based for _tests_ selecting with `data` attribute.

Reply:
well we could just convert the aria values in getActiveCell to return the 0 based index if it is just used there.
After we can argue that, it could be useful to add the data-colIndex directly on the element 🤔

Reply:
As for `data-rowIndex`, it is used here as well (not only tests):
https://github.com/mui-org/material-ui-x/blob/ca7718b7f1afbbf056f00309d691094f75f5c978/packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts#L73.
So the question is would it be fine to add `data-colIndex` just for _tests_?

> well we could just convert the aria values in getActiveCell to return the 0 based index if it is just used there.

Why decrement aria by one for tests (currently it is 1-based)? I think it would be confusing later on.

Reply:
let's add a return type in `getActiveCell `, we have `GridCellIndexCoordinates` that seems like a good fit.
Index coordinates are 0 based. So the function signature will be self explanatory.

> Why decrement aria by one for tests (currently it is 1-based and should be 1-based)? I think it would be confusing later on.

Currently it is 0 based as there is this issue, and we use data-rowindex.

> So the question is would it be fine to add data-colIndex just for tests?
Not only for tests, it will also be for consistency so you can go ahead I think...

Reply:
@dtassone Do you mean changing assertions if considering return type `GridCellIndexCoordinates`? 
```suggestion
    expect(getActiveCell()).to.deep.equal({ colIndex: 0, rowIndex: 0 });
```

Reply:
Ah yes it's quite verbose.
```tsx
expect(formatCoordinates(getActiveCell())).to.deep.equal('0-0'); 
expect(formatIndexes(getActiveCell())).to.deep.equal('0-0'); 
```

What do you think? 🤔

Reply:
Does not look worth to me. If we are considering `format` for coordinates, then what is the benefit of having signature for `getActiveCell` after wrapping it around `format`?  `getActiveCell` is a test util only. Instead would keep `getActiveCell` as it is. What do you think? I am not sure. 

This is how it would be then:
```ts
function formatCooridnates(cellCoordinates: GridCellIndexCoordinates) {
  const { rowIndex, colIndex } = cellCoordinates;
  return `${rowIndex}-${colIndex}`;
}

export function getActiveCell(): GridCellIndexCoordinates | null {
  let activeElement: Element | null;
  if (document.activeElement && document.activeElement.getAttribute('role') === 'cell') {
    activeElement = document.activeElement;
  } else {
    activeElement = document.activeElement && document.activeElement.closest('[role="cell"]');
  }

  if (!activeElement) {
    return null;
  }

  const cellCoordinates = {
    rowIndex: Number(activeElement.getAttribute('data-rowindex')),
    colIndex: Number(activeElement.getAttribute('data-colindex')),
  };

  return cellCoordinates;
}
```

Note: We also have `getActiveColumnHeader`

Reply:
Fine. Let's keep it as a string, just use the correct values for indexes. 
No need to enforce the return type...

We could add a test to make sure that getActiveCell works as expected... 🤔

Reply:
Updated. Considered `data-colindex`. Not much changes in tests now after removal of considering `aria-colindex`.